pkg/configurator/crd_methods.go

```diff
@@ -0,0 +1,161 @@
+package configurator
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+const (
+	// defaultServiceCertValidityDuration is the default validity duration for service certificates
+	defaultServiceCertValidityDuration = 24 * time.Hour
+)
+
+// The functions in this file implement the configurator.Configurator interface
+
+// GetOSMNamespace returns the namespace in which the OSM controller pod resides.
+func (c *CRDClient) GetOSMNamespace() string {
+	return c.osmNamespace
+}
+
+func marshalConfigToJSON(config *osmConfig) ([]byte, error) {
+	return json.MarshalIndent(config, "", "    ")
+}
+
+// GetMeshConfig returns the MeshConfig in pretty JSON.
+func (c *CRDClient) GetMeshConfig() ([]byte, error) {
+	cm, err := marshalConfigToJSON(c.getMeshConfig())
+	if err != nil {
+		log.Error().Err(err).Msgf("Error marshaling MeshConfig %s: %+v", c.getMeshConfigCacheKey(), c.getMeshConfig())
+		return nil, err
+	}
+	return cm, nil
+}
+
+// IsPermissiveTrafficPolicyMode tells us whether the OSM Control Plane is in permissive mode,
+// where all existing traffic is allowed to flow as it is,
+// or it is in SMI Spec mode, in which only traffic between source/destinations
+// referenced in SMI policies is allowed.
+func (c *CRDClient) IsPermissiveTrafficPolicyMode() bool {
+	return c.getMeshConfig().PermissiveTrafficPolicyMode
```

Comment:
unrelated: now and since you reworked all this, it'd be good to cache state so every config call doesn't have to parse the entire config map every time. We just need to update values when they change.

Reply:
This is similar to what @shashankram  mentioned, that we can remove the intermediate `osmconfig` struct. Will work on it in a separate PR.